### ISSUE TITLE
[Input] The error prop does not allow links

### ISSUE THREAD

**Issue by theetrain** [telus/tds-core#894]
## Description

Originally reported by @ah-arch 

Defect: When making use of the core-input component's `error` prop, I pass paragraph text with a core-link component, then a prop-type error appears since `error` only accepts strings.

Expectation: I can pass paragraph text and a core-link component without receiving a prop-type error.

Note, this defect may also be applicable to other form components.

## Reproduction Steps

1. Within an application's development environment, insert an Input component that makes use of the `error` prop like this:
   ```jsx
    <Input label="First name" value="Harry" error={(
      <Paragraph>Hello <Link href="https://google.com">What is this</Link></Paragraph>
    )} />
   ```
2. Observe a propType error with for the `error` prop within the FormField component

## Reproduction Percentage (if applicable)

- 10/10 times

## Meta

- TDS component version: @tds/core-input@1.0.13 
- Willing to develop solution: Yes @ah-arch 
- Has workaround: Yes 
- High impact: No 

## Screenshots

![image](https://user-images.githubusercontent.com/12798751/51347281-a72ccf80-1a6d-11e9-9568-a199c61e3569.png)

**Comment by ah-arch** [comment]
The `Notification`component already automatically adds a `p` tag, and therefore it would be invalid to have a nested `p` tag in the `p` tag, or `div`s.
So the only allowed elements are really `span` or `<React.Fragment>{string} <Link /></React.Fragment>`
This is our scenario using a `Fragment`:
![error with link](https://user-images.githubusercontent.com/4450690/51347592-8ca72600-1a6e-11e9-89c4-6d6f57f7ecfa.png)

**Comment by Christina-Lo** [comment]
There is consensus that this is a useful pattern from the DPA meeting and we will move forward with implementing this

**Comment by ah-arch** [comment]
Realized that the `error` prop was not placed in a `Notification`, but an `InputFeedback` with a nested `Paragraph` instead.

Leaving it as it, and just updating `error` to `PropTypes.oneOfType([PropTypes.string, PropTypes.node])`.

**Comment by marcod1419** [comment]
@ah-arch That sounds good! Just one thing, `PropTypes.node` could be `componentWithName(Link)`. If you're not aware of it, we recently created a [PropTypes utility](https://tds.telus.com/components/index.html#proptypes) so we have a lightweight way of being more specific with our props. Thanks so much!!

**Comment by ah-arch** [comment]
Most likely the node will contain a combination of strings and `Link`.

Is this wording okay for **Usage criteria**?
> We recommend `React.Fragment` when supplying custom strings to the `error` prop; `div` is not valid in a `p` tag

**Comment by theetrain** [comment]
This has been resolved in [@tds/core-input@2.1.0](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-input%402.1.0)
